gateway/header.go

```diff
@@ -80,12 +80,9 @@ func HeaderN(ctx context.Context, key string, n int) (val []string, found bool)
 	}
 }
 
-// PrefixOutgoingHeaderMatcher prefixes outgoing gRPC metadata with
-// runtime.MetadataHeaderPrefix ("Grpc-Metadata-").
-// It behaves like the default gRPC-Gateway outgoing header matcher
-// (if none is provided as an option).
+// PrefixOutgoingHeaderMatcher discards all grpc header metadata.
 func PrefixOutgoingHeaderMatcher(key string) (string, bool) {
-	return fmt.Sprintf("%s%s", runtime.MetadataHeaderPrefix, key), true
+	return "", false
```

Comment:
Looks like this way we completely remove an ability to pass data from grpc layer to user. For reported bug this seems to work fine, but how we are going to handle cases when this is actually required?

Reply:
This is the default OutgoingHeaderMatcher. One is able to implement its own and apply to the gateway.